error.go

```diff
@@ -149,14 +149,14 @@ var ErrNoData = internal.ErrNoData
 
 // NewNonRetryableApplicationError creates new instance of non-retryable *ApplicationError with reason and optional details.
 // Use ApplicationError for any use case specific errors that cross activity and child workflow boundaries.
-func NewNonRetryableApplicationError(reason string, details ...interface{}) *ApplicationError {
-	return internal.NewApplicationError(reason, true, details...)
+func NewNonRetryableApplicationError(reason string, cause error, details ...interface{}) *ApplicationError {
```

Comment:
Add comment that cause can be nil

Reply:
Or better add Method NewApplicationErrorWithCause and remote cause from this one

Reply:
rename reason to type